10.1 RN (#6318)
mendix/docs: content/en/docs/releasenotes/studio-pro/10/10.1.md

---
title: "10.1"
url: /releasenotes/studio-pro/10.1/
description: "The release notes for Mendix Studio Pro version 10.1 (including all patches) with details on new features, bug fixes, and known issues."
weight: 99
# KI: WOR-1563
---

## 10.1 {#101} 

**Release date: July 25, 2023**

> 

Reviewer: We will need the button inserted here, something like
```suggestion

{{% button color="info" href="https://marketplace.mendix.com/link/studiopro/10.1.0" text="Go to Marketplace" title="Download version 10.1.0 from the Marketplace" %}}

```

* We added **readonly** description fields in the **Add** and **Edit** dialog boxes for [business event subscriptions](/appstore/modules/business-events/).
* We improved the configuration of parameter mappings for actions and snippet calls. This prevents them from changing automatically when you add a page parameter or snippet parameter to the surrounding form, or when you change a data grid's selection mode.
* The target document of an action or snippet call no longer requires the parameter mappings to be closed and reopened to be refreshed if they have become invalid.
* We added an ability to preview images in the **Properties** pane for entity and building block images.
* We now publish the [OpenAPI 3.0 specification](/refguide/open-api/) for published REST services alongside the existing OAS or Swagger 2.0 specification.
* The Swagger UI of published REST services now shows a **Choose File** button when you are using a REST operation that takes a FileDocument (or derived entity) as a parameter.

#### Other Improvements

> * We made the workflows behavior for user tasks more consistent. If the number of targeted users is fewer than the required number of users, then the user task will always fail.

Reviewer: I think this might be a case where it is "less than" as we are talking about the number, not the users themselves?

There are fewer users, but the number is less?
But, not so important!

Author: I believe the correct term is "fewer", because the **number** of targeted users is still a countable noun. I know it looks a little strange, but I think it's right.

> * We have updated spacing between the toolbar control elements.

Reviewer: ```suggestion
* We have updated the spacing between the toolbar control elements.
```
?? Could be either

Author: Good change

> * We updated the `mx check` tool so it refers to the rebranded Best Practice Bot (formerly Performance Bot).

Reviewer: UX element?
```suggestion
* We updated the `mx check` tool so it refers to the rebranded **Best Practice Bot** (formerly Performance Bot).
```

> * We enabled more translations of error messages generated by the Mendix Runtime. 

Author: I will get more detail on which languages were added into translation options for this?

### Fixes

* <a id="wo-ki-fix"></a>We fixed a [known issue](/releasenotes/studio-pro/10.0/#wo-ki) where the algorithm for detecting conflicts during the migration of workflows with running multi-user tasks is currently too rigid and will not allow a conflicted user task to be continued in certain cases.
* <a id="pane-ki-fix"></a>We fixed a [known issue](/releasenotes/studio-pro/10.0/#pane-ki) by restoring the ability to change the **Visible for** page setting from the **Properties** pane. (Ticket 189995)
* We fixed an issue where the XPath constraint of a data source could produce incorrect results if the widget was placed inside a pluggable widget and the constraint was using attributes from a page variable (such as `$currentObject`). (Ticket 178210)
* We fixed an issue in offline-first profiles where a failed synchronzation due to committing invalid changes provided insufficient information about the error in both the error logs and the stack trace. (Tickets 183328, 186301)

> * We fixed an issue where no consistency error was shown on an invalid expression in a microflow when using an association that cannot be traversed. (Ticket 185701)

Reviewer: for?
```suggestion
* We fixed an issue where no consistency error was shown for an invalid expression in a microflow when using an association that cannot be traversed. (Ticket 185701)
```

Author: Good change

* We fixed the behavior of the [Pluggable Widget Filtering API](/apidocs-mxsdk/apidocs/pluggable-widgets-client-apis-list-values/#listvalue-filtering) for attributes over association. Now, a filter such as `equals(attributeId, empty())` will also match if any of the associations leading up to the attribute are empty. (Ticket 186042)
* We fixed an issue where [Validation Assist](/refguide/validation-assist/) generated an incorrect validation microflow. (Ticket 186222)
* We fixed an issue that allowed users to add custom icons to native pages even though native pages do not support custom icons. (Ticket 186783)
* We fixed an error which occurred when calling `Core.http().executeHttpRequests()` from a Java action, and when a **Services** log level is set to **TRACE**. (Tickets 187764 and 187228)

> * We fixed an issue in offline applications where an association would be lost if an uncommitted NPE would be used in a microflow. (Ticket 187904)

Reviewer: I think we should spell this out. I think NPE is an internal shorthand, not in our documentation? (See next bullet point)
```suggestion
* We fixed an issue in offline applications where an association would be lost if an uncommitted non-persistable entity were to be used in a microflow. (Ticket 187904)
```

> * We fixed an issue when retrieving a non-persistable entity over association and the entity has a generalization with a calculated attribute. (Ticket 188255)

Reviewer: ```suggestion
* We fixed an issue when retrieving a non-persistable entity over association if the entity has a generalization with a calculated attribute. (Ticket 188255)
```

* We fixed an issue which caused an error when deploying an app if a snippet contained a check box with an **On change** action which used a snippet parameter. (Ticket 191010)
* We made it possible to switch off localization (time zone conversion) for DateTime attributes of external entities (Ticket 188260).
* We fixed an issue where fallback to the default language from non-default languages was not used for a [text template property](/apidocs-mxsdk/apidocs/pluggable-widgets-property-types/#texttemplate) when it was linked to a data source. This resulted in an empty text in the client (for example, in the dynamic text property of a [Data Grid 2](/appstore/modules/data-grid-2/) column). (Tickets 188724, 188403)
* We fixed an issue where Studio Pro would crash if you accidentally dropped a [container widget](/refguide/container/) onto itself. (Ticket 189191)
* We fixed an issue where creating a non-persistable entity could lead to errors when attributes were read-only.

> * We fixed an issue in Best Practice Bot where apps could report 0 recommendations for an MXP008 cached non-persistable entity.

Reviewer: MXP008? I think this needs to be simpler for users who are not familiar with this bot:
```suggestion
* We fixed an issue in Best Practice Bot where apps might not make any recommendations even if a non-persistable entity is cached (recommendation [MXP008](/refguide/performance-best-practices/#mxp008)).
```

> * We fixed an issue where the Machine Learning Kit had an issue when an app is running.

Reviewer: Two many issues?

> * We fixed the documentation links in Studio Pro to point to Mx10 docs (instead of Mx9 docs).

Reviewer: HOORAY!!!!!!

Author: lolllllll

* We fixed an issue where the Mendix Runtime for published REST services would give an error when an operation microflow would set a non-standard HTTP status code for the HTTP response.

> * We fixed an issue affecting [text boxes](/refguide/text-box/) where the enter keypress event was being called from popups, causing an unexpected behavior.

Reviewer: Do we need an article here?
```suggestion
* We fixed an issue affecting [text boxes](/refguide/text-box/) where the enter keypress event was being called from popups, causing unexpected behavior.
```

Author: Yup. It's ideas like this that make you the genuine article.

> * We fixed icons styling in the auto-complete popup of the text editor control in the dark theme.

Reviewer: ```suggestion
* We fixed icon styling in the auto-complete popup of the text editor control in the dark theme.
```

Author: Correct, my mistake.

* We fixed an issue with rendering of the custom application icons in the app selector dialog box.
* We fixed an issue where a tabbed editor (for example, a navigation editor) that contains a tree or grid control would start changing its width upon performing UI updates (for example, toggling a checkbox) after the appearance of a scrollbar in that same tree or grid control.
* We fixed an issue where the change event handler of a completed user task was not called when an incompatible workflow was continued.
* We fixed an error that caused an exception in workflows change event handling when the context object did not exist.
* We fixed an issue that could cause an error message when closing a dialog box.
* We fixed an issue with the combobox and toggle controls background colors.

> * We fixed an issue with the request body URLs encoding for [POST and PATCH requests](/refguide/published-rest-operation/) for external objects. These URLs will now be encoded for the outgoing request's body.

Reviewer: "For" seems wrong here - I would say that a URL is "in" the request body?
```suggestion
* We fixed an issue with the request body URLs encoding for [POST and PATCH requests](/refguide/published-rest-operation/) for external objects. These URLs will now be encoded in the outgoing request's body.
```

Author: This seems right.

* We fixed an accessibility issue affecting the search bar in list view component.

> * We fixed an issue related to input labels accessibility in login widgets.

Author: I will try to find these specific "login widgets" and link them.

Author: Changed to the correct name and linked.

* We fixed an issue where the XPath constraint of a data source could produce incorrect results if the widget was placed inside a pluggable widget and the constraint was using attributes from a page variable (such as `$currentObject`). (Ticket 178210)
* We fixed an issue where Studio Pro would crash if you accidentally dropped a container widget onto itself. (Ticket 189191)
* We fixed an issue which caused an error when deploying an app if a snippet contained a check box with an **On change** action which used a snippet parameter. (Ticket 191010)
* We fixed an issue where a tabbed editor (for example, a navigation editor) that contains a tree or grid control would start changing its width upon performing UI updates (for example, toggling a checkbox) after the appearance of a scrollbar in that same tree or grid control.
* We fixed an issue where the change event handler of a completed user task was not called when an incompatible workflow was continued.
* We fixed an issue that could cause an error message when closing a dialog box.

> * We fixed an accessibility issue on the search bar in list view component.

Author: Check "list view component" language.

> * We fixed an issue with the combobox and toggle controls background colors.

Author: What is a combobox? I don't see it documented anywhere.

Should this be combobox and toggle control's background colors?

Reviewer: I can't find this anywhere - although there is a community module about comboboxes.
Have we got an answer from FrontEnd yet?

Author: Apparently these are not widgets (though there used to be a combobox widget), but Studio Pro UI things. I am checking with the CTRL team.

Author: I got details here, I'll include them somehow tomorrow: https://mendix.slack.com/archives/C0295GTBG12/p1690208285137989?thread_ts=1690207209.948219&cid=C0295GTBG12
